### DISCUSSION TITLE
About the S3 gateway implementation

### DISCUSSION THREAD

**Discussion by 0xWrapper** [treeverse/lakeFS#8115] [General]
I am learning your code, I would like to ask     the pkg/gateway is on the S3 gateway of all the code?

**Comment by arielshaqed** [comment]
Hi @0xWrapper ,
Sorry we missed this.  I'm actually not sure how I can help, so I will give some general guidelines and you can ask again if they are insufficiently detailed.  The [S3 Gateway](https://docs.lakefs.io/reference/s3.html) allows you to use any S3 client to read and write with lakeFS, by setting its endpoint URL.  Package gateway handles all S3 operations and translates them into Graveler and other operations as needed.
I hope this helps!
